internal/provider/providers/porkbun/provider.go

```diff
@@ -119,16 +119,31 @@ func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Add
 		recordType = constants.AAAA
 	}
 	ipStr := ip.String()
-	recordIDs, err := p.getRecordIDs(ctx, client, recordType)
+	records, err := p.getRecords(ctx, client, recordType)
 	if err != nil {
 		return netip.Addr{}, fmt.Errorf("getting record IDs: %w", err)
 	}
 
-	if len(recordIDs) == 0 {
-		// ALIAS record needs to be deleted to allow creating an A record.
-		err = p.deleteALIASRecordIfNeeded(ctx, client)
-		if err != nil {
-			return netip.Addr{}, fmt.Errorf("deleting ALIAS record if needed: %w", err)
+	if len(records) == 0 {
+		// For new domains, Porkbun Creates 2 Default DNS Records which point to their "parked" domain page:
+		// ALIAS domain.tld -> pixie.porkbun.com
+		// CNAME *.domain.tld -> pixie.porkbun.com
+		// ALIAS and CNAME records conflict with A and AAAA records, and attempting to create an A or AAAA record
+		// will return a 400 error if they aren't first deleted.
+		porkbunParkedDomain := "pixie.porkbun.com"
+		switch {
+		case p.owner == "@":
+			// Delete ALIAS domain.tld -> pixie.porkbun.com record
+			err = p.deleteMatchingRecord(ctx, client, constants.ALIAS, porkbunParkedDomain)
```

Comment:
We should make sure this deletes only for the `@` owner

Reply:
Are you thinking we should pass in the owner we want to delete for? p.deleteMatchingRecord uses p.getRecords which uses p.owner, so as it's coded currently, this could only be for the  `@` owner